### PR TITLE
SHRINKRES-277 Upgraded to maven-invoker 3.0.0

### PR DESCRIPTION
**Pull request by MatousJobanek**

### CODE REVIEW

File: maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/ConfigurationStageTestCase.java [hunk 1]
```diff
@@ -81,12 +80,13 @@ public void runTest() {
         assertEquals(shellEnvironments, invocationRequest.getShellEnvironments());
         assertEquals(threads, invocationRequest.getThreads());
         assertEquals(toolChainsFile, invocationRequest.getToolchainsFile());
+        assertEquals(globalToolChainsFile, invocationRequest.getGlobalToolchainsFile());
         assertEquals(userSettingFile, invocationRequest.getUserSettingsFile());
-        assertEquals(true, invocationRequest.isActivatedReactor());
+        assertEquals(builderId, invocationRequest.getBuilder());
```
Comment (bartoszmajsak) [inline maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/ConfigurationStageTestCase.java]: One handy tip here would be to use `assertj` with its `soft-assertions`. This way you would have a chance to gather all the failed conditions in one test execution rather than discovering them one after the other.

Comment (bartoszmajsak) [inline maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/ConfigurationStageTestCase.java]: Also maybe worth to split this test to multiple ones if we test more than one aspect of the build here?

File: maven/impl-maven-embedded/pom.xml [hunk 2]
```diff
@@ -14,7 +14,6 @@
     <description>Implementation for running Maven builds and retrieving information and archives from the built project</description>
 
     <properties>
-        <version.maven.invoker>2.2</version.maven.invoker>
```
Comment (bartoszmajsak) [inline maven/impl-maven-embedded/pom.xml]: We could update spacelift too ;)

Comment (MatousJobanek) [inline maven/impl-maven-embedded/pom.xml]: will apply on all properties - will be part of next commit

File: maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/ConfigurationStageTestCase.java (hunk 1 above)
Comment (MatousJobanek) [inline maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/ConfigurationStageTestCase.java]: soft assertions make more sense...

File: maven/impl-maven-embedded/pom.xml (hunk 2 above)
Comment (MatousJobanek) [inline maven/impl-maven-embedded/pom.xml]: next commit - not part of this PR

**Comment by bartoszmajsak** [comment]
Any idea why the build is failing?

**Comment by MatousJobanek** [comment]
nope, no idea

**Comment by MatousJobanek** [comment]
Thx for review @bartoszmajsak 
Landed in https://github.com/shrinkwrap/resolver/commit/c51623023949d84f77377e417837ab87bce49e97
